[#129] Make management entrypoint tests shared

rinn7e opened the pull request:
## Description

Problem: Management tests between Lorentz and LIGO version are very
similar. We should create shared tests to use between them instead to
reduce duplication.

Solution: Make management tests shared between Lorentz and LIGO.

<!--
Describes the nature of your changes. If they are substantial, you should
further subdivide this into a section describing the problem you are solving and
another describing your solution.
-->

## Related issue(s)

<!--
- Short description of how the PR relates to the issue, including an issue link.
For example
- Fixed #100500 by adding lenses to exported items

Write 'None' if there are no related issues (which is discouraged).
-->

Resolves #129 

## :white_check_mark: Checklist for your Pull Request

<!--
Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

If you don't set a checkmark (e. g. don't add a test for new functionality),
you must be able to justify that.
-->

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

rinn7e commented:
Hope to get your review on this as well @pasqu4le .

pasqu4le commented on ligo/haskell/test/Test/Ligo/BaseDAO/Management.hs:
```diff
@@ -245,34 +144,4 @@ test_BaseDAO_Management =
-  :: (MonadNettest caps base m)
-  => m a -> m ()
-expectForbiddenXTZ = expectCustomError_ #fORBIDDEN_XTZ
+      ! defaults
```
Missing an end-of-file newline

pasqu4le commented on src/BaseDAO/ShareTest/Management.hs:
```diff
@@ -0,0 +1,296 @@
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+--
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+-- For all the incomplete list pattern matches in the calls to with
+-- withOriginatedFn func
```
Nitpick here as well.
```suggestion
-- For all the incomplete list pattern matches in the calls to the
-- `withOriginatedFn` function
```

pasqu4le commented on test/Test/BaseDAO/Management.hs:
```diff
@@ -3,7 +3,7 @@
 --
 {-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
 -- For all the incomplete list pattern matches in the calls to with
--- withOriginated func
+-- withOriginated initialStorage func
```
Same nitpick as above

pasqu4le commented on test/Test/BaseDAO/Management.hs:
```diff
@@ -12,16 +12,39 @@ module Test.BaseDAO.Management
+import Lorentz.Contracts.BaseDAO.Types
+import BaseDAO.ShareTest.Management
+
+-- | Function that originates the contract and also make a bunch of
```
```suggestion
-- | Function that originates the contract and also makes a bunch of
```

rinn7e replied:
I've made all the changes you suggested now.

pasqu4le reviewed (commented):
Only minor nitpicks, otherwise LGTM

pasqu4le reviewed (commented):
One more thing: in the issue title and description, as well as in the commit message you spelled `managment` which seems to be and incorrect form of `management`, please fix them.

Feel free to squash the commits as well

rinn7e commented:
> One more thing: in the issue title and description, as well as in the commit message you spelled `managment` which seems to be and incorrect form of `management`, please fix them.
> 
> Feel free to squash the commits as well

Yeah, right, I didn't check it properly, I've squashed and fixed the typo now, you can check this, Pasquale.
